system/Entity.php

```diff
@@ -163,7 +164,21 @@ public function toArray(bool $onlyChanged = false, bool $cast = true): array
 				continue;
 			}
 
-			$return[$key] = $this->__get($key);
+			$value = $this->__get($key);
```

Comment:
Is it necessary to use an additional variable here? IMO `$return[$key]` would work just fine.